src/braket/circuits/serialization.py

```diff
@@ -32,6 +33,31 @@ class QubitReferenceType(str, Enum):
     PHYSICAL = "PHYSICAL"
 
 
+class SerializableProgram(ABC):
+    @abstractmethod
+    def to_ir(
+        self,
+        ir_type: IRType = IRType.OPENQASM,
+        build_if_necessary: bool = False,
+    ) -> str:
+        """Serializes the program into an intermediate representation.
+
+        Args:
+            ir_type (IRType): The IRType to use for converting the program to its
+                IR representation. Defaults to IRType.OPENQASM.
+            build_if_necessary (bool): Whether to allow the program to be implicitly
```

Comment:
This is a little weird. Why do we need it?

Reply:
Agreed, this arg doesn't belong here. Removing it.